### PR TITLE
Migrate code to support Python 3

### PR DESCRIPTION
**Pull request by Suor**
Made it work both in Python 2 and 3. Doctests were painful :)

### CODE REVIEW

**Comment by kmike** [comment]
Looks great!

Yeah, doctests are painful. In NLTK I've solved in by creating a custom nose [plugin](https://github.com/nltk/nltk/blob/develop/nltk/test/doctest_nose_plugin.py) which considers `"foo"` and `u"foo"` the same in the doctest output; maybe it is possible to do something similar for pytest. Doctests are used to build API docs, so 'bprint' and 'bformat' make docs less clear. But I guess fixing it is out of scope now.

Could you please rename cross.py to compat.py? It is a more usual name for such modules, and it has less meanings (e.g. one can think cross.py is something related to cross product).

**Comment by Suor** [comment]
Done. Also, webstruct uses nose, so you could apply your magic.

**Comment by Suor** [comment]
Good reason to publish your plugin separately )

**Comment by kmike** [comment]
Heh, we're using nosetests indeed :) The plugin in NLTK is quite messy, but I guess it is ok to publish it as-is. I've also found https://pypi.python.org/pypi/doctest-ignore-unicode/ - if you have some time, could you please check if it can replace bprint / bformat?

We should also fix docs (install instructions in intro.rst) and setup.py (classifiers).

**Comment by Suor** [comment]
I will probably have some time tomorrow. I have almost zero experience with
nose though.

**Comment by Suor** [comment]
Indeed I managed to get rid of `bprint()/bformat()` with a help of plugin. I still needed to call `.decode()` on bytestrings since they are represented differently too. I added and issue to plugin about that - https://github.com/gnublade/doctest-ignore-unicode/issues/1, so it could be made even smoother in future.

**Comment by kmike** [comment]
Great! 

What do you think about using +IGNORE_UNICODE by default, as well as +ELLIPSIS? Nose doctest plugin (http://nose.readthedocs.org/en/latest/plugins/doctests.html) has --doctest-options argument which can be used to set default doctest flags.

File: webstruct/feature_extraction.py [hunk 1]
```diff
@@ -166,12 +168,12 @@ def tokenize_single(self, tree):
             >>> html_tokenizer = HtmlTokenizer(replace_html_tags={'b': 'strong'})
             >>> tree = loader.loadbytes(b"<p>hello, <PER>John <b>Doe</b></PER> <br> <PER>Mary</PER> said</p>")
             >>> html_tokens, tags = html_tokenizer.tokenize_single(tree)
-            >>> html_tokens  # doctest: +ELLIPSIS
-            [HtmlToken(token=u'hello', parent=<Element p at ...>, index=0), HtmlToken...]
-            >>> tags
-            ['O', u'B-PER', u'I-PER', u'B-PER', 'O']
+            >>> print(html_tokens[0].token, html_tokens[0].parent.tag)  # doctest: +ELLIPSIS
```
Comment (kmike) [inline webstruct/feature_extraction.py]: is this print still needed?

Comment (Suor) [inline webstruct/feature_extraction.py]: It's not. Updated this as well.

**Comment by Suor** [comment]
On global +IGNORE_UNICODE and +ELLIPSIS - it's your call.

**Comment by kmike** [comment]
It is almost ready to merge. 

1. Let's add +IGNORE_UNICODE and +ELLIPSIS to global flags (it'll make docstrings nicer);
2. setup.py classifiers should be fixed;
3. intro.rst should mention that Python 3.3+ is also supported.

**Comment by Suor** [comment]
Done all three.

**Comment by kmike** [comment]
:+1:

**Pull request merged**